spec/generators/install_generator_spec.rb

```diff
@@ -18,35 +20,59 @@ module Admin
           class ApplicationController < Administrate::ApplicationController
       RB
     end
+
+    it "uses the namespace option" do
+      stub_generator_dependencies
+      controller = file("app/controllers/manager/application_controller.rb")
+
+      run_generator ["--namespace", "manager"]
+
+      expect(controller).to exist
+      expect(controller).to have_correct_syntax
+      expect(controller).to contain <<-RB.strip_heredoc
+        module Manager
+          class ApplicationController < Administrate::ApplicationController
+      RB
+    end
   end
 
   describe "routes generator" do
     it "invokes the routes generator if there are no namespace resources" do
-      begin
-        stub_generator_dependencies
-        Rails.application.routes.draw {}
+      stub_generator_dependencies
+      Rails.application.routes.draw {}
 
-        run_generator
+      run_generator
 
-        expect(Rails::Generators).to invoke_generator("administrate:routes")
-      ensure
-        reset_routes
-      end
+      expect(Rails::Generators).to invoke_generator(
+        "administrate:routes", ["--namespace", "admin"]
+      )
     end
 
     it "does not invoke routes generator if namespace routes already exist" do
-      begin
-        stub_generator_dependencies
-        Rails.application.routes.draw do
-          namespace(:admin) { resources :customers }
-        end
+      stub_generator_dependencies
+      Rails.application.routes.draw do
+        namespace(:admin) { resources :customers }
+      end
 
-        run_generator
+      run_generator
 
-        expect(Rails::Generators).not_to invoke_generator("administrate:routes")
-      ensure
-        reset_routes
+      expect(Rails::Generators).not_to invoke_generator(
+        "administrate:routes", ["--namespace", "admin"]
+      )
+    end
+
+    it "does not invoke routes generator if namespace routes already exist \
+    when using custom namespace" do
```

Comment:
Do you think we'd be able to use a `context` here, instead of having a super long `it`?